### PR TITLE
Feat/fe 84 change channel to private

### PR DESCRIPTION
**Pull request by queen-x-vee**
# Make channel private functionality/(https://linear.app/team-bevel/issue/FRO-84/change-channel-to-private-functionality)

( This could be an existing issue or a linear ticket )

**My PR Closes #issue_number_here** or
**My PR Fixes ID-linear_ticket_number_here**

# Changes proposed

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

[video display.webm](https://user-images.githubusercontent.com/62246293/204140100-6c497372-b67e-4e46-b110-98187c560234.webm)

### CODE REVIEW

File: packages/main/src/auth/useChannel.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+const useChannel = channel_id => {
+  const BASE_API_URL = "https://chat.zuri.chat";
+  const token = sessionStorage.getItem("token");
+  //const member_id = localStorage.getItem("user");
+  const org_id = localStorage.getItem("currentWorkspace");
+  const member = JSON.parse(sessionStorage.getItem("user")) || null;
+  const member_id = member.id;
```
Comment (billmal071) [inline packages/main/src/auth/useChannel.js:9]: If it's null, how do you want to get the id?

File: packages/main/src/auth/useChannel.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  const createChannel = data => {
+    //const aisosa= `${BASE_API_URL}/api/v1/org/${org_id}/members/${member_id}`
+    //console.log(data,aisosa, "here");
+    axios.post(
+      `${BASE_API_URL}/api/v1/org/${org_id}/members/${member_id}/rooms`,
+      data,
+      {
+        headers: {
+          //Authorization: `Bearer ${token}`,
```
Comment (billmal071) [inline packages/main/src/auth/useChannel.js:20]: Remove commented out code, also are you sure token doesn't need to be sent in the request?

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx [hunk 3]
```diff
@@ -58,6 +61,7 @@ function MessageRoomDetailsDialog({
   const toggleLeaveChannelModal = () =>
     setShowLeaveChannelModal(!showLeaveChannelModal);
   const toggleArchiveChannel = () => setShowArchiveChannel(!showArchiveChannel);
+  const togglePrivateChannel = () => setShowPrivateChannel(!showPrivateChannel);
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx:64]: do this instead. setShowArchiveChannel(prev => !prev), same for setShowPrivateChannel.

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx [hunk 4]
```diff
@@ -31,6 +32,7 @@ import FileList from "./components/FileList";
 // getting edited topic from local storage
 const getEditedTopicFromLS = () => {
   let editedTopics = localStorage.getItem("editedTopic");
+  console.log(editedTopics);
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx:35]: Remove all console.log.

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx [hunk 5]
```diff
@@ -529,7 +538,15 @@ function SettingPanel({
       <ChannelWrapper>
         <Channels>
           <AiOutlineLock />
-          <ChannelContent>Change to Private Channel</ChannelContent>
+          {/*<ChannelContent>Change to Private Channel</ChannelContent>*/}
+          <Typography
+            onClick={() => {
+              // closeModal()
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx:544]: remove commented out code.

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 6]
```diff
@@ -24,9 +24,7 @@ const PrivateChannel = ({ closeEdit }) => {
 
   const privateChannel = () => {
     const tina = user?.find(x => x.id == org_id)?.member_id;
-    console.log(user);
-    console.log(`hi ${tina}`);
-    console.log(`hello ${room}`);
+    console.log("loading");
     if (data !== null) {
       const newData = { ...data, is_private: true };
       console.log(newData);
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:30]: remove console.log

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 7]
```diff
@@ -24,9 +24,7 @@ const PrivateChannel = ({ closeEdit }) => {
 
   const privateChannel = () => {
     const tina = user?.find(x => x.id == org_id)?.member_id;
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:26]: use a triple ===.

File: packages/main/src/components/protected/sidebar/components/DropRoom.jsx [hunk 8]
```diff
@@ -1,10 +1,26 @@
...
+import useChannel from "../../../../auth/useChannel";
 
 const DropRoom = props => {
+  const { updatePrivacy, createChannel } = useChannel(props.data?.roomId);
+  const [state, setState] = useState({});
+
+  const handleEvent = ({ target }) => {
+    setState({
+      ...state,
+      [target.name]: target.type === "checkbox" ? target.checked : target.value
```
Comment (billmal071) [inline packages/main/src/components/protected/sidebar/components/DropRoom.jsx:15]: setState(prev => ({ ..prev, then update the values of what you need to change.}))

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 9]
```diff
@@ -7,7 +7,8 @@ const PrivateChannel = ({ closeEdit }) => {
   const org_id = localStorage.getItem("currentWorkspace");
   const user = JSON.parse(sessionStorage.getItem("organisations"));
   const room = window.location.href.split("/").at(6);
-  const BASE_URL = "https:chat.zuri.chat";
+  const BASE_URL = "https://chat.zuri.chat";
+  //const BASE_URL = "https://chat.zuri.chat";
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:11]: remove commented out code.

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 10]
```diff
@@ -33,7 +37,7 @@ const PrivateChannel = ({ closeEdit }) => {
         )
         .then(res => {
           console.log(res);
-          window.location.reload();
+          //window.location.reload();
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:38]: remove console.log and commented out code.

File: packages/main/src/components/top-navigation-bar/TopNavigationBar.module.css [hunk 11]
```diff
@@ -114,7 +115,7 @@
 .navbarBrand span {
   padding: 0.5rem;
   font-size: 22px;
-  margin: 1rem 0.5rem;
+  /* margin: 1rem 0.5rem; */
```
Comment (billmal071) [inline packages/main/src/components/top-navigation-bar/TopNavigationBar.module.css:118]: remove.

**Review by billmal071 — CHANGES_REQUESTED** [review]

File: packages/main/src/auth/useChannel.js (hunk 1 above)
Comment (queen-x-vee) [reply packages/main/src/auth/useChannel.js]: I'm no longer making use of the file, deleted it.

File: packages/main/src/auth/useChannel.js (hunk 2 above)
Comment (queen-x-vee) [reply packages/main/src/auth/useChannel.js]: deleted file

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx (hunk 3 above)
Comment (queen-x-vee) [reply packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx]: done

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx (hunk 4 above)
Comment (queen-x-vee) [reply packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx]: done

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx (hunk 5 above)
Comment (queen-x-vee) [reply packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx]: Done

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx (hunk 7 above)
Comment (queen-x-vee) [reply packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx]: Done

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx (hunk 6 above)
Comment (queen-x-vee) [reply packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx]: Done

File: packages/main/src/components/top-navigation-bar/TopNavigationBar.module.css (hunk 11 above)
Comment (queen-x-vee) [reply packages/main/src/components/top-navigation-bar/TopNavigationBar.module.css]: Done

Comment (queen-x-vee) [reply packages/main/src/components/top-navigation-bar/TopNavigationBar.module.css]: Done

**Comment by billmal071** [comment]
@queen-x-vee please pull again and push.

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 12]
```diff
@@ -17,25 +16,22 @@ const PrivateChannel = ({ closeEdit }) => {
...
         setData(res.data.data);
-        //console.log(`response is ${JSON.stringify(res.data.data)}`)
       })
       .catch(e => console.log(e));
   }, [5]);
 
   const privateChannel = () => {
-    const tina = user?.find(x => x.id == org_id)?.member_id;
+    const tina = user?.find(x => x.id === org_id)?.member_id;
     console.log("loading");
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:25]: remove console.log.

File: packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx [hunk 13]
```diff
@@ -17,25 +16,22 @@ const PrivateChannel = ({ closeEdit }) => {
...
       axios
         .put(
           `${BASE_URL}/api/v1/org/${org_id}/members/${tina}/rooms/${room}`,
           newData
         )
         .then(res => {
           console.log(res);
-          //window.location.reload();
         })
         .catch(e => console.log(e));
```
Comment (billmal071) [inline packages/ui/src/message-room-details-dialog/components/PrivateChannel.jsx:36]: remove console.log and use a toast notfication to show alerts or notifications.

File: packages/main/src/components/protected/sidebar/components/ToggleSwitch.jsx [hunk 14]
```diff
@@ -1,7 +1,50 @@
...
+      const newData = { ...data, is_private: true };
+      console.log(newData);
+      axios
+        .put(
+          `${BASE_URL}/api/v1/org/${org_id}/members/${tina}/rooms/${room}`,
+          newData
+        )
+        .then(res => {
+          console.log(res);
+          //window.location.reload();
```
Comment (billmal071) [inline packages/main/src/components/protected/sidebar/components/ToggleSwitch.jsx]: remove.

File: packages/main/src/components/protected/sidebar/components/ToggleSwitch.jsx [hunk 15]
```diff
@@ -1,7 +1,50 @@
...
+      })
+      .catch(e => console.log(e));
+  }, [5]);
+
+  const privateChannel = () => {
+    const tina = user?.find(x => x.id == org_id)?.member_id;
+    console.log("loading");
+    if (data !== null) {
+      const newData = { ...data, is_private: true };
+      console.log(newData);
```
Comment (billmal071) [inline packages/main/src/components/protected/sidebar/components/ToggleSwitch.jsx:33]: remove.

File: packages/main/src/components/protected/sidebar/components/ToggleSwitch.jsx [hunk 16]
```diff
@@ -1,7 +1,50 @@
...
+    setState(!state);
+    console.log(state);
+  };
+
+  useEffect(() => {
+    axios
+      .get(`${BASE_URL}/api/v1/org/${org_id}/rooms/${room}`)
+      .then(res => {
+        setData(res.data.data);
+        //console.log(`response is ${JSON.stringify(res.data.data)}`)
```
Comment (billmal071) [inline packages/main/src/components/protected/sidebar/components/ToggleSwitch.jsx]: remove logs and commented out code.

**Review by billmal071 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**